app/models/articles.models.js

```diff
@@ -7,17 +7,23 @@ exports.selectArticleById = (articleId) => {
 	});
 };
 
-exports.selectAllArticles = () => {
-	return db
-		.query(
-			`SELECT articles.article_id, articles.title, articles.topic, articles.author, articles.created_at, articles.votes, articles.article_img_url, COUNT(comments.article_id)::INT AS comment_count
+exports.selectArticles = (topic) => {
+	const queries = [];
+
+	let sqlString = `SELECT articles.article_id, articles.title, articles.topic, articles.author, articles.created_at, articles.votes, articles.article_img_url, COUNT(comments.article_id)::INT AS comment_count
   FROM articles 
-  LEFT JOIN comments 
-  ON articles.article_id = comments.article_id
-  GROUP BY articles.article_id
-  ORDER BY articles.created_at DESC`
-		)
-		.then(({ rows }) => rows);
+	LEFT JOIN comments 
+  ON articles.article_id = comments.article_id `;
+
+	if (topic) {
+		sqlString += `WHERE topic = $1 `;
+		queries.push(topic);
+	}
+
+	sqlString += `GROUP BY articles.article_id
+	ORDER BY articles.created_at DESC`;
+
+	return db.query(sqlString, queries).then(({ rows }) => rows);
```

Comment:
You may need to adjust your query to line up with the comments on your tests here.